x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/ForceMergeAction.java

```diff
@@ -28,42 +30,53 @@
 public class ForceMergeAction implements LifecycleAction {
     public static final String NAME = "forcemerge";
     public static final ParseField MAX_NUM_SEGMENTS_FIELD = new ParseField("max_num_segments");
+    public static final ParseField BEST_COMPRESSION_FIELD = new ParseField("best_compression");
 
     private static final ConstructingObjectParser<ForceMergeAction, Void> PARSER = new ConstructingObjectParser<>(NAME,
         false, a -> {
         int maxNumSegments = (int) a[0];
-        return new ForceMergeAction(maxNumSegments);
+        boolean bestCompression = a[1] != null && (boolean) a[1];
+        return new ForceMergeAction(maxNumSegments, bestCompression);
     });
 
     static {
         PARSER.declareInt(ConstructingObjectParser.constructorArg(), MAX_NUM_SEGMENTS_FIELD);
+        PARSER.declareBoolean(ConstructingObjectParser.constructorArg(), BEST_COMPRESSION_FIELD);
     }
 
     private final int maxNumSegments;
+    private final boolean bestCompression;
```

Comment:
I wonder about instead of exposing this as best_compressiom true/false, it would be clearer to expose this as `codec`, and if unspecified we don’t set index.codec, and otherwise set the codec to the value specified (only allowing for best_compression). The reason this appeals to me as otherwise we have an impedance mismatch between how it’s configured on the index level versus how it’s configured in ILM. By avoiding the impedance mismatch now, we let the two evolve together, and also make it easier for users to go between the two worlds. I’m curious to hear your thoughts on this?

Reply:
I think I'd also prefer to expose this as `codec` to match the index level configuration name.

Reply:
@jasontedor & @dakrone: Thanks for your input. I can work on this and let you know. 👍

Reply:
@dakrone /@jasontedor  In this case, we don't need the UpdateSettingsStep? Please let me know if my understanding is rite?

Thanks.

Reply:
We'd still need the `UpdateSettingsStep`, and it would still update the `index.codec` setting. What Jason is saying here is that the force_merge step should allow specifying a `"codec"` parameter, which then a user could set to `best_compression` similar to the way someone would specify it were they setting it in the [index settings](https://www.elastic.co/guide/en/elasticsearch/reference/7.5/index-modules.html#_static_index_settings). This has less mismatch than having a "best_compression" parameter that is set to true or false in the ILM configuration.